src/Roslyn.Diagnostics.Analyzers/CSharp/BlankLines/CSharpBlankLinesBetweenStatementsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Diagnostics.Analyzers;
+
+namespace Roslyn.Diagnostics.CSharp.Analyzers.BlankLines
+{
+    /// <summary>
+    /// Analyzer that finds code of the form:
+    /// <code>
+    /// if (cond)
+    /// {
+    /// }
+    /// NextStatement();
+    /// </code>
+    /// 
+    /// And requires it to be of the form:
+    /// <code>
+    /// if (cond)
+    /// {
+    /// }
+    /// 
+    /// NextStatement();
+    /// </code>
+    /// 
+    /// Specifically, all blocks followed by another statement must have a blank line between them.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpBlankLinesBetweenStatementsDiagnosticAnalyzer : DiagnosticAnalyzer
```

Comment:
In roslyn-analyzers, it was preferred that all analyzers (even language-specific) be implemented as abstract classes in the Core layer to exist in the proper tracking analyzers file.

Tagging @mavasani to confirm.

Reply:
Yep that was the decision, see https://github.com/dotnet/roslyn-analyzers/pull/4564#discussion_r545292242

Reply:
i'm fine with that.  i was plannign on doing a similar analyzer/fixer for VB as well.